taskvine/src/worker/vine_cache.c

```diff
@@ -250,6 +250,18 @@ int vine_cache_queue_transfer(struct vine_cache *c, const char *source, const ch
 	return 1;
 }
 
+/*
+Perform a remote file transfer to produce a file.
+*/
+
+int vine_cache_transfer_now(struct vine_cache *c, const char *source, const char *cachename, int64_t size, int mode)
+{
+	struct vine_cache_file *f = vine_cache_file_create(VINE_CACHE_TRANSFER, source, size, mode, 0);
+	hash_table_insert(c->table, cachename, f);
+	vine_cache_ensure(c, cachename);
```

Comment:
Suggest that you just make this method a variant of `vine_cache_queue_transfer` and have an "immediate" flag of some kind.